collector.php

```diff
@@ -55,9 +55,9 @@
             echo "OK\n";
         } else {
             // Set the 'blockeddomain' and 'blockedurlpath' values.
-            $parsedurl = parse_url($blockeduri);
-            $blockeddomain = $parsedurl['host'];
-            $blockedurlpath = $parsedurl['path'];
+            $parsedurl = new \moodle_url($blockeduri);
```

Comment:
Can we always safely assume this is a moodle URI?  It feels correct to have some level of parsing and sanity checking in place here, as previous

Reply:
I found out in the SEB work that despite it being called moodle_url, it actually just does a lot of useful parsing to any url you pass it. We use it to break apart safe exam browser urls. 

Reason I used it here is that if a 'host' isn't found in the URI, it won't throw an error and just provide back the default which is an empty string. Previously it wouldn't save a record as an exeption would be thrown